spec/requests/admin_controller_spec.rb

```diff
@@ -38,6 +38,12 @@
   end
 
   describe "#repos" do
+    before do
+      # Return only the first 3 repos to reduce the number of calls to 'git' CLI
+      repos = DockerManager::GitRepo.find_all
+      DockerManager::GitRepo.stubs(:find_all).returns(repos[0...3])
```

Comment:
Nitpick, but when selecting X items from an array, I usually prefer the `array[0, 3]` or `.take(3)` syntax to avoid any "off by one" errors.